backend/src/bundles/users/user.repository.ts

```
@@ -158,6 +174,122 @@ class UserRepository implements Repository {
     public delete(): ReturnType<Repository['delete']> {
         return Promise.resolve(true);
     }
+
+    public async addFollowing(
+        id: number,
+        followingId: number,
+    ): Promise<UserFollowingsResponseDto | null> {
+        const trx = await this.userModel.startTransaction();
+        try {
+            const user = await this.userModel.query(trx).findById(id);
+            if (!user) {
+                return null;
+            }
```

With this approach, transactions may time out before they are closed, leaving an open connection to the database. It is better to throw an error to promptly close the transaction.

---

Thanks, I fixed it